--- a/keystone/frontends/d5_compat.py
+++ b/keystone/frontends/d5_compat.py
@@ -247,9 +247,10 @@
             else:
                 raise fault.IdentityFault("%s not initialized with data" % \
                                           self.__class__.__str__)
-        d5_data = {"auth": {}}
-        for key, value in self.json.iteritems():
-            d5_data["auth"][key] = value
+        d5_data = {"auth": self.json.copy()}
+        d5_data['auth']['serviceCatalog'] = \
+            dict([(s['type'], s['endpoints'])
+                   for s in d5_data['auth']['serviceCatalog']])
 
         return json.dumps(d5_data)
 
