--- a/keystone/manage/__init__.py
+++ b/keystone/manage/__init__.py
@@ -130,7 +130,7 @@
             print "SUCCESS: User %s disabled." % object_id
 
     elif (object_type, command) == ('user', 'list'):
-        print_table(('id', 'enabled', 'tenant'), api.list_users())
+        print_table(('id', 'name', 'enabled', 'tenant'), api.list_users())
 
     elif (object_type, command) == ('tenant', 'add'):
         if api.add_tenant(name=object_id):
--- a/keystone/manage/api.py
+++ b/keystone/manage/api.py
@@ -28,7 +28,7 @@
     objects = db_api.USER.get_all()
     if objects == None:
         raise IndexError("No users found")
-    return [[o.id, o.enabled, o.tenant_id] for o in objects]
+    return [[o.id, o.name, o.enabled, o.tenant_id] for o in objects]
 
 
 def add_tenant(name):
