--- a/keystone/test/sampledata.py
+++ b/keystone/test/sampledata.py
@@ -31,8 +31,8 @@
         'http://nova.publicinternets.com/v1.0/',
         'http://127.0.0.1:8774/v1.0', 'http://localhost:8774/v1.0', '1', '0'),
     ('endpointTemplates', 'add', 'RegionOne', 'nova',
-        'http://nova.publicinternets.com/v1.1/', 'http://127.0.0.1:8774/v1.1',
-        'http://localhost:8774/v1.1', '1', '0'),
+        'http://nova.publicinternets.com/v1.1/%tenant_id%', 'http://127.0.0.1:8774/v1.1/%tenant_id%',
+        'http://localhost:8774/v1.1/%tenant_id%', '1', '0'),
     ('endpointTemplates', 'add', 'RegionOne', 'glance',
         'http://glance.publicinternets.com/v1.1/%tenant_id%',
         'http://nova.admin-nets.local/v1.1/%tenant_id%',
@@ -50,8 +50,8 @@
         'http://nova.publicinternets.com/v1.0/', 'http://127.0.0.1:8774/v1.0',
         'http://localhost:8774/v1.0', '1', '0'),
     ('endpointTemplates', 'add', 'RegionOne', 'compute_v1',
-        'http://nova.publicinternets.com/v1.1/', 'http://127.0.0.1:8774/v1.1',
-        'http://localhost:8774/v1.1', '1', '0'),
+        'http://nova.publicinternets.com/v1.1/%tenant_id%/', 'http://127.0.0.1:8774/v1.1/%tenant_id%',
+        'http://localhost:8774/v1.1/%tenant_id%', '1', '0'),
     ('endpointTemplates', 'add', 'RegionOne', 'image',
         'http://glance.publicinternets.com/v1.1/%tenant_id%',
         'http://nova.admin-nets.local/v1.1/%tenant_id%',
